packages/rlc-common/src/buildSamples.ts

```diff
@@ -0,0 +1,835 @@
+import { getClientName } from "./helpers/nameConstructors.js";
+import { NameType, camelCase, normalizeName } from "./helpers/nameUtils.js";
+import {
+  RLCModel,
+  RLCSampleGroup,
+  File as RLCFile,
+  Paths,
+  OperationMethod,
+  RLCSampleDetail,
+  SampleParameters,
+  SampleParameter,
+  PathMetadata,
+  OperationParameter,
+  ObjectSchema,
+  SchemaContext,
+  Schema,
+  ArraySchema,
+  DictionarySchema
+} from "./interfaces.js";
+import { sampleTemplate } from "./static/sampleTemplate.js";
+// @ts-ignore: to fix the handlebars issue
+import hbs from "handlebars";
+import * as path from "path";
+import { isObjectSchema } from "./helpers/schemaHelpers.js";
+import { getImmediateParentsNames } from "./buildObjectTypes.js";
+
+export function buildSamples(model: RLCModel) {
```

Comment:
Can we have unit test for this file?  Feel free to refactor the design if needed.

Reply:
Sure, I have a pr to enable the UT in rlc-common project. https://github.com/Azure/autorest.typescript/pull/2022